src/components/ControlledTextInputField/ControlledTextInputField.js

```diff
@@ -26,12 +26,22 @@ const ControlledTextInputField = ({
   name,
   label,
   placeholder,
-  rules,
+  customValidationFunction,
   error,
   required,
 }) => {
   const classes = useStyles()
 
+  const validateRequired = (value) => {
+    if (required && !value) return "This field is required"
+
+    const nonSpaceRegex = /\S/
+    if (required && !nonSpaceRegex.test(value))
+      return "This field should not be only space"
```

Comment:
This looks like something that might be useful elsewhere too, you should make a validation function just for this